circuit/list.go

```diff
@@ -0,0 +1,77 @@
+package circuit
+
+// a simple list to keep breakers sorted by access order.
+// It can return the first consecutive items that match a
+// condition (in our case, the ones that were inactive for a
+// while).
+type list struct {
```

Comment:
I think this is the first time ever since university, that I see a linked list implementation :-)

Anyway. Why not just use an array? This seems like premature optimisation and just adds fuckup-surface. Please consider just using an array. Yes, it might be slower, no it won't matter